tensornetwork/network.py

```diff
@@ -1,1083 +0,0 @@
-# Copyright 2019 The TensorNetwork Authors
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-"""Implementation of TensorNetwork structure."""
-
-import h5py
-# pylint: disable=line-too-long
-from typing import Any, Sequence, List, Set, Optional, Union, Text, Tuple, Type, Dict, BinaryIO
-import numpy as np
-from tensornetwork import config
-#pylint: disable=useless-import-alias
-import tensornetwork.network_components as network_components
-#pylint: disable=useless-import-alias
-import tensornetwork.network_operations as network_operations
-from tensornetwork.backends import backend_factory
-import logging
-
-Tensor = Any
-string_type = h5py.special_dtype(vlen=str)
-
-
-class TensorNetwork:
-  """Implementation of a TensorNetwork."""
-
-  def __init__(self,
-               backend: Optional[Text] = None,
-               dtype: Optional[Type[np.number]] = None) -> None:
-    """
-    Args:
-      backend (str): name of the backend. Currently supported 
-                     backends are 'numpy', 'tensorflow', 'pytorch', 'jax', 'shell'
-      dtype: dtype of the backend of network. If `None`, no dtype checks 
-             are performed. Default value is `None`. For backend 
-             initialization functions like `zeros`, `ones`, `randn` a 
-             dtype of `None` defaults to float64
-    """
-    logging.warn(
-        "The TensorNetwork object has been DEPRECATED and will be removed "
-        "in future releases.\n"
-        "Updating your code is fairly easy. For a detailed tutorial, please "
-        "visit "
-        "https://medium.com/@keeper6928/upgrading-your-tensornetwork-code-b032f0ab3dd4"
-    )
-    if backend is None:
-      backend = config.default_backend
```

Comment:
Maybe instead of totally nuking `TensorNetwork`, we have it raise a `DeprecationWarning` https://docs.python.org/3/library/exceptions.html.

This gives users a slightly cleaner heads up. It will also link to the upgrade post and help them port their code.

Reply:
Yeah, fair enough. However, some parts of the code still won't work, i.e. contractors only accept iterables of nodes in this PR. Perhaps then we should not have nuked anything and deprecated everything :)?

Reply:
What's wrong with contractors just accepting an iterable of `Node`s? I've been using the for personal research for a while and I haven't run into any issues. (though sometimes I use a `tn.reachable(edge.node1)` hack, that should be fixed with #332).

Reply:
Nothing wrong, on the contrary is the only supported way in this PR. What I'm concerned about is that we confuse users by
1. not nuking the TensorNetwork class, but adding a deprecation warning
2. nuking some of the functionalities around the TensorNetwork object (like being able to pass them to a contractor) which the PR is removing without any warning.

Reply:
Aw I see your concern. The `DeprecationWarning` is actually an exception in python (terrible name, but we don't have control over that). All of the functionality around and inside of `TensorNetwork` will be removed.

Basically, I only want `TensorNetwork` to look like this:

```python
class TensorNetwork():
  def __init__(...):
    raise DeprecationWarning("The TensorNetwork object has been...")
```
So when someone updates their tensornetwork version and tries to run their code, it should immediately crash with this message:
```python
>>> tn.TensorNetwork()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  ...
DeprecationWarning: The TensorNetwork object has been...
```

Make sense? Sorry for the confusion.

Reply:
Makes sense! I was indeed thinking about a deprecation decorator that leaves functionality intact for now.

Reply:
Added this